src/control/server/server_utils.go

```diff
@@ -182,7 +182,9 @@ func prepBdevStorage(srv *server, iommuEnabled bool, hpiGetter common.GetHugePag
 	}
 	// The config value is intended to be per-engine, so we need to adjust
 	// based on the number of engines.
-	prepReq.HugePageCount = srv.cfg.NrHugepages * len(srv.cfg.Engines)
+	if srv.cfg.NrHugepages > 0 {
+		prepReq.HugePageCount = srv.cfg.NrHugepages * len(srv.cfg.Engines)
```

Comment:
```suggestion
		if len(srv.cfg.Engines) == 0 {
		        prepReq.HugePageCount = srv.cfg.NrHugepages
		} else {
		        prepReq.HugePageCount = srv.cfg.NrHugepages * len(srv.cfg.Engines)
		}
```
To cover the case where nr_hugepages has been set in the config file but no engines have been defined.